neqo-http3/src/connection_client.rs

```diff
@@ -549,7 +549,18 @@ impl Http3Client {
                 }
                 ConnectionEvent::RecvStreamReadable { stream_id } => {
                     if let Err(e) = self.handle_stream_readable(stream_id) {
-                        if e.stream_reset_error() {
+                        let is_critical =
+                            if let Some(r) = self.base_handler.recv_streams.get(&stream_id) {
+                                matches!(
+                                    r.stream_type(),
+                                    Http3StreamType::Control
+                                        | Http3StreamType::Encoder
+                                        | Http3StreamType::Decoder
+                                )
+                            } else {
+                                false
+                            };
```

Comment:
You might consider moving this to a `stream_is_critical(StreamId)` function on the base handler.